sssd_test_framework/utils/override.py

```diff
@@ -0,0 +1,375 @@
+"Managing local users and groups."
+
+from __future__ import annotations
+
+import jc
+from pytest_mh import MultihostHost, MultihostUtility
+from pytest_mh.cli import CLIBuilder, CLIBuilderArgs
+from pytest_mh.ssh import SSHLog
+
+__all__ = [
+    "OverrideUtils",
+    "OverrideUser",
+    "OverrideGroup",
+]
+
+
+class OverrideUtils(MultihostUtility[MultihostHost]):
+    """
+    Management of override users and groups, sss_override.
+
+    .. note::
+
+        All changes are automatically reverted when a test is finished.
+    """
+
+    def __init__(self, host: MultihostHost) -> None:
+        """
+        :rtype: object
+        :param host: Remote host instance.
+        :type host: MultihostHost
+        """
+        super().__init__(host)
+
+        self.cli: CLIBuilder = CLIBuilder(host.ssh)
+        self._users: list[str] = []
+        self._groups: list[str] = []
+
+    def teardown(self) -> None:
+        """
+        Teardown
+        """
+        super().teardown()
+
+    def user(self, name: str) -> OverrideUser:
+        """
+        Get override user object.
+
+        .. code-block:: python
+            :caption: Example usage
+
+            @pytest.mark.topology(KnownTopology.Client)
+            def test_example(client: Client):
+                # Create override
+                client.override.users('user1').add(name='o-user1', uid=10001, gid=10001, gecos="test")
+
+                # Assert the overridden name, uid, gid and gecos
+                assert client.tools.getent.passwd("o-user1").uid == 10001
+                assert client.tools.getent.passwd("o-user1").gid == 10001
+                assert client.tools.getent.passwd("o-user1").gecos == "test"
+
+        :param name: User name.
+        :type name: str
+        :return: New user override object.
+        :rtype: OverrideUser
+        """
+        return OverrideUser(self, name)
+
+    def group(self, name: str) -> OverrideGroup:
+        """
+        Get override group object.
+
+        .. code-block:: python
+            :caption: Example usage
+
+            @pytest.mark.topology(KnownTopology.Client)
+            def test_example(client: Client):
+                # Create override
+                client.override.group('users').add(name='o-users', gid=10001)
+
+                # Assert the overridden name and gid
+                assert client.tools.getent.group("o-users").gid == 10001
+
+        :param name: Group name.
+        :type name: str
+        :return: New group object.
+        :rtype: OverrideGroup
+        """
+        return OverrideGroup(self, name)
+
+
+class OverrideUser(object):
```

Comment:
I think that implicit inheritance from object is not needed in python 3:
https://www.w3docs.com/snippets/python/why-do-python-classes-inherit-object.html

Reply:
That is true, but rest of the code also use object explicitly.